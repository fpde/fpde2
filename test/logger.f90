program logger_test

   use logger_module

   type(named) :: o

   print *, "Current logger level: ", logger%log_level
   print *, "Current logger path: ", "'"//trim(logger%path)//"'"

   call set_log_level(FPDE_LOG_ERROR)
   print *, "Current logger level: ", logger%log_level

   ! call get_new_logfile_unit(o%logfile_unit,"log.log")
   ! print *, "Object logfile unit", o%logfile_unit

   logger%path="./"
   print *, "Current logger path: ", "'"//trim(logger%path)//"'"

   ! o%logfile_unit=0
   print *, "Object logfile unit", o%logfile_unit

   call call_logs(o)

contains

   subroutine call_logs(o)
      type(named) :: o

      call o%log(FPDE_LOG_ERROR,"this is fancy error log message")
      call o%log(FPDE_LOG_WARNING,"this is fancy warning message")
      call o%log(FPDE_LOG_INFO,"this is fancy info message")
      call o%log(FPDE_LOG_DEBUG,"this is fancy debug message")
   
   end subroutine call_logs

end program logger_test
