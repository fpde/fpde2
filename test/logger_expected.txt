 Logger level:                      1
 Logger path: './'
 Object logfile unit:                     6
 
 Calling logs with logger level:                      1
 20120308-235917.851: ERROR     : this is a fancy error log message
 
 Calling logs with logger level:                      2
 20120308-235917.851: ERROR     : this is a fancy error log message
 20120308-235917.851: WARNING   : this is a fancy warning message
 
 Calling logs with logger level:                      3
 20120308-235917.851: ERROR     : this is a fancy error log message
 20120308-235917.851: WARNING   : this is a fancy warning message
 20120308-235917.851: INFO      : this is a fancy info message
 
 Calling logs with logger level:                      4
 20120308-235917.851: ERROR     : this is a fancy error log message
 20120308-235917.851: WARNING   : this is a fancy warning message
 20120308-235917.851: INFO      : this is a fancy info message
 20120308-235917.851: DEBUG     : this is a fancy debug message
